pkg/server/entry.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2021 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package server
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"github.com/tektoncd/chains/proto"
+)
+
+func getEntry(uid string) (*proto.Entry, error) {
+	log.Printf("Getting entry with uid %s...", uid)
+	f := path(uid)
+	contents, err := ioutil.ReadFile(f)
+	if err != nil {
+		return nil, errors.Wrapf(err, "getting entry for %s", uid)
+	}
+	var e proto.Entry
+	if err := json.Unmarshal(contents, &e); err != nil {
+		return nil, errors.Wrap(err, "unmarshal")
+	}
+	return &e, nil
+}
+
+func addEntry(e *proto.Entry) error {
+	log.Printf("Adding entry %v...", e)
+	contents, err := json.Marshal(e)
+	if err != nil {
+		return errors.Wrap(err, "marshal binary")
+	}
+	f := path(e.Uid)
+	fmt.Println(f)
+	if err := ioutil.WriteFile(f, contents, 0644); err != nil {
```

Comment:
The `ioutil.WriteFile` makes it hard to unit test. Could this be instead a https://pkg.go.dev/io#Writer? which would be easier to mock and test?

Reply:
i added in a unit test which shouldn't require changing this!